### PR TITLE
Fix package distribution & change output dir to `dist`

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "allowJs": true,
     "allowSyntheticDefaultImports": true,
     "alwaysStrict": true,
+    "declaration": true,
```
Comment (refi93) [inline tsconfig.base.json:6]: question - does this flag enable exporting the type definitions? Does this mean we weren't exporting the types until now? Asking to understand if I need to make another pre-release so Adalite can make use of ledgerjs public types

**Review by refi93 — APPROVED** [review]

File: tsconfig.base.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "allowJs": true,
```
Comment (refi93) [inline tsconfig.base.json:3]: I guess we are fully migrated to ts by now, so if this flag was to allow some interoperability with legacy js code, we should be able to drop it

File: tsconfig.base.json (hunk 1 above)
Comment (ppershing) [reply tsconfig.base.json:6]: Yep, exactly, we weren't generating .d.ts files

File: tsconfig.base.json (hunk 2 above)
Comment (ppershing) [reply tsconfig.base.json]: done

**Pull request merged**